### PR TITLE
[Fix] Respect Spark2.5 attention gate activation mode

### LINKED ISSUE #37639: [Bug] Spark2.5 attention output gate hardcodes sigmoid, ignores config gate_attn_act_mode
### Checklist

- [x] I searched open and closed issues/PRs and found no existing report for this mismatch.
- [x] The mismatch is present on the latest `main` commit.
- [x] A minimal CPU-only reproduction is included below.
- [x] This is a bug report rather than a general question.
- [x] This report is written in English.

### Describe the bug

The Spark2.5 attention output gate applies a per-head scalar gate computed by `g_proj`, then activates it with a hardcoded `torch.sigmoid`:

[`python/sglang/srt/models/spark2_5.py`](https://github.com/sgl-project/sglang/blob/f8cbf000f4a5bfd86d3fb7c1e2d6c8fb12339d0e/python/sglang/srt/models/spark2_5.py#L198-L206)

```python
if self.headwise_attn_output_gate:
    g, _ = self.g_proj(hidden_states)
    g = torch.sigmoid(g.float()).to(attn_output.dtype)
    gate_output = attn_output.view(...) * g.unsqueeze(-1)
    attn_output = gate_output.view(*attn_output.shape)
```

The activation function is **not** hardcoded in the reference. The official `modeling_spark.py` selects it from the config field `gate_attn_act_mode`, supporting `sigmoid` and `silu`, and raises on anything else:

[`modeling_spark.py`](https://huggingface.co/XHToken/Spark-X2.5-4B/blob/main/modeling_spark.py) (lines 199-204)

```python
if self.gate_attn_act_mode == "sigmoid":
    gate = torch.sigmoid(gate_score.float())
elif self.gate_attn_act_mode == "silu":
    gate = F.silu(gate_score.float())
else:
    raise ValueError(f"Unsupported gate_attn_act_mode: {self.gate_attn_act_mode}")
```

The two functions differ substantially: `sigmoid(x) ∈ (0, 1)` is bounded and strictly positive, while `silu(x) = x · sigmoid(x)` is unbounded and goes negative for `x < 0`. Because the gate is multiplied into every head of every attention layer's output, the error accumulates layer over layer and the model output diverges from the reference. SGLang also has no `else: raise`, so an unsupported mode would fail silently rather than erroring at load time.

### Current impact

Both released checkpoints — [Spark-X2.5-4B](https://huggingface.co/XHToken/Spark-X2.5-4B/blob/main/config.json) and [Spark-X2.5-1.7B](https://huggingface.co/XHToken/Spark-X2.5-1.7B/blob/main/config.json) — set `gate_attn_act_mode: "sigmoid"`, so the hardcoded value happens to match and there is **no numerical divergence on the released models today**. I verified this by reading both configs.

This is a latent/forward-compatibility mismatch rather than an active regression: any future Spark checkpoint declaring `gate_attn_act_mode: "silu"` (the reference explicitly supports it) would be served with the wrong gate activation and produce garbage output, with no error to point at the cause. It is the same class of "hardcode-the-known-config-value, drop the config read" omission as #37609 (GELU approximation), but in the attention path.

### Reproduction

The activation divergence is visible on CPU without loading model weights:

```python
import torch
import torch.nn.functional as F

x = torch.linspace(-6, 6, 20001, dtype=torch.float32)

# SGLang hardcoded path
sglang_gate = torch.sigmoid(x)

# reference path for a "silu" checkpoint
ref_gate = F.silu(x)

print("SGLang (hardcoded sigmoid) range:", sglang_gate.min().item(), sglang_gate.max().item())
print("reference (silu) range:        ", ref_gate.min().item(), ref_gate.max().item())
print("max abs diff:                  ", (sglang_gate - ref_gate).abs().max().item())
```

Observed output:

```text
SGLang (hardcoded sigmoid) range: 0.002472617670893669 0.997527361869812
reference (silu) range:         -0.3171774744987488 6.0
max abs diff:                   5.4314165115356445
```

For the released `sigmoid` checkpoints the two paths coincide (`max abs diff = 0.0`), consistent with "no current divergence, latent mismatch".

### Expected behavior

Read `gate_attn_act_mode` from the config in `Spark2_5Attention.__init__` and branch on it in `forward`, mirroring the reference's three branches including the `else: raise`. Something like:

```python
# __init__
self.gate_attn_act_mode = getattr(config, "gate_attn_act_mode", "sigmoid")

# forward
if self.gate_attn_act_mode == "sigmoid":
    g = torch.sigmoid(g.float())
elif self.gate_attn_act_mode == "silu":
    g = F.silu(g.float())
else:
    raise ValueError(f"Unsupported gate_attn_act_mode: {self.gate_attn_act_mode}")
g = g.to(attn_output.dtype)
```

### Environment

Reported against `main` at `f8cbf000f4a5bfd86d3fb7c1e2d6c8fb12339d0e`. No GPU run needed — the mismatch is in the activation function selection, verifiable from config + source alone.

### PR DESCRIPTION
**Pull request by ActiveSky**
## Motivation

Fixes #37639. Spark2.5 attention output gating currently always applies
`sigmoid`, even when a checkpoint declares a different supported
`gate_attn_act_mode`. This makes future `silu` checkpoints diverge silently
from the reference implementation.

## Modifications

- Read `gate_attn_act_mode` from the Spark2.5 config.
- Support the reference `sigmoid` and `silu` activations.
- Raise a clear error for unsupported activation modes.
- Add CPU unit coverage for both modes and invalid configuration.

The existing `sigmoid` default preserves the behavior of currently released
Spark2.5 checkpoints.

## Validation

- Focused Spark2.5 CPU tests added.

Full model/GPU validation was not available in this environment.

## Checklist

- [x] Format code with the repository pre-commit configuration.
- [x] Add focused regression tests.
- [x] No documentation update is required.
- [x] No speed benchmark is required.

---
### CI States

Latest PR Test (Base): :x: [Run #33722427014](https://github.com/sgl-project/sglang/actions/runs/33722427014)
Latest PR Test (Extra): :x: [Run #33722426984](https://github.com/sgl-project/sglang/actions/runs/33722426984)
Latest PR Test (AMD ROCm 7.2): :x: [Run #33722427092](https://github.com/sgl-project/sglang/actions/runs/33722427092)

### CODE REVIEW

**Comment by ActiveSky** [comment]
Hi maintainers,

There has been no formal review activity on this PR for some time, so we wanted to check whether this change is still relevant and whether we should continue working on it. We understand priorities can change; if the issue is already covered elsewhere or the scope should be adjusted, please let us know and we will follow your guidance.

If the PR is still useful, could you please take a look when you have time? Feedback on the implementation, tests, or whether the branch should be updated would be very helpful. We hope to contribute in a way that helps move the SGLang project forward.

Thank you very much for your time.
